core/src/Streamly/Internal/Data/Fold.hs

```diff
@@ -434,6 +437,50 @@ scan (Fold stepL initialL extractL) (Fold stepR initialR extractR) =
 
     extract = extractR . snd
 
+-- | Scan the input of a 'Fold' to change it in a stateful manner using another
+-- 'Fold'. If second fold terminates it will restart again from initial state.
+-- /Pre-release/
+scanMany :: Monad m => d -> Fold m a b -> Refold m d b c -> Fold m a c
```

Comment:
I am not sure about what would be a useful way to implement this using Refold. Can you implement it using Fold only, that is much simpler to understand and use?